suite/performance/common.go

```diff
@@ -99,3 +108,246 @@ func recordResults(b ginkgo.Benchmarker, ID string, results []time.Duration) err
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+
+	if _, ok := r.results[requestID]; !ok {
+		r.results[requestID] = &PerfJobRunResult{}
+	}
+	return r.results[requestID]
+}
+
+// GetAll returns all test results
+func (r *PerfRequestIDTestResults) GetAll() map[string]*PerfJobRunResult {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
+	return r.results
+}
+
+func (r *PerfRequestIDTestResults) setResultStartTimes(clients []client.Chainlink, jobMap ContractsNodesJobsMap) error {
+	g := errgroup.Group{}
+	for contract := range jobMap {
+		contract := contract
+		g.Go(func() error {
+			return r.setResultStartTimeByContract(clients, jobMap, contract)
+		})
+	}
+	return g.Wait()
+}
+
+func (r *PerfRequestIDTestResults) setResultStartTimeByContract(clients []client.Chainlink, jobMap ContractsNodesJobsMap, contract interface{}) error {
+	for _, chainlink := range clients {
+		jobRuns, err := chainlink.ReadRunsByJob(jobMap[contract][chainlink].GetJobID())
+		if err != nil {
+			return err
+		}
+		log.Debug().
+			Str("Node", chainlink.URL()).
+			Int("Runs", len(jobRuns.Data)).
+			Msg("Total runs")
+		for _, jobDecodeData := range jobRuns.Data {
+			rqInts, err := actions.ExtractRequestIDFromJobRun(jobDecodeData)
+			if err != nil {
+				return err
+			}
+			rqID := common.Bytes2Hex(rqInts)
+			loc, _ := time.LoadLocation("UTC")
+			startTime := jobDecodeData.Attributes.CreatedAt.In(loc)
+			log.Debug().
+				Time("StartTime", startTime).
+				Str("RequestID", rqID).
+				Msg("Request found")
+			d := r.Get(rqID)
+			d.StartTime = startTime
+		}
+	}
+	return nil
+}
+
+func (r *PerfRequestIDTestResults) calculateLatencies(b ginkgo.Benchmarker) error {
+	var latencies []time.Duration
+	for rqID, testResult := range r.GetAll() {
+		latency := testResult.EndTime.Sub(testResult.StartTime)
+		log.Debug().
+			Str("RequestID", rqID).
+			Time("StartTime", testResult.StartTime).
+			Time("EndTime", testResult.EndTime).
+			Dur("Duration", latency).
+			Msg("Calculating latencies for request id")
+		if testResult.StartTime.IsZero() {
+			log.Warn().
+				Str("RequestID", rqID).
+				Msg("Start time zero")
+		}
+		if testResult.EndTime.IsZero() {
+			log.Warn().
+				Str("RequestID", rqID).
+				Msg("End time zero")
+		}
+		if latency.Seconds() < 0 {
+			log.Warn().
+				Str("RequestID", rqID).
+				Msg("Latency below zero")
+		} else {
+			latencies = append(latencies, latency)
+		}
+	}
+	if err := recordResults(b, "Request latency", latencies); err != nil {
+		return err
+	}
+	return nil
+}
+
+// NodeData common node data
+type NodeData interface {
+	GetJobID() string
+	GetProvingKeyHash() [32]byte
+}
+
+// RunlogNodeData node data required for runlog test
+type RunlogNodeData struct {
+	JobID string
+}
+
+// GetJobID gets internal job id
+func (n RunlogNodeData) GetJobID() string {
+	return n.JobID
+}
+
+// GetProvingKeyHash gets proving key hash for VRF
+func (n RunlogNodeData) GetProvingKeyHash() [32]byte {
+	return [32]byte{}
```

Comment:
Why is this empty? Does runlog not actually have one of these?

Reply:
Nope, kinda ugly thing, but I hope sometimes we can separate metrics collecting from test logic/deployment completely

Reply:
Will fix a comment though